src/Actions/Transactions.js

```diff
@@ -71,43 +70,41 @@ export const getReceipts = function(transactions) {
 export const SET_BLOCK_REQUESTED = `${prefix}/SET_BLOCK_REQUESTED`
 export const ADD_TRANSACTIONS_TO_VIEW = `${prefix}/ADD_TRANSACTIONS_TO_VIEW`
 export const getTransactionsForBlock = function(number) {
-  return function(dispatch, getState) {
+  return async function(dispatch, getState) {
     let requested = getState().transactions.blocksRequested
 
     // If it's already requested, bail
     if (requested[number] === true) {
       return
     }
 
-    let provider = getState().web3.provider
+    let web3Instance = getState().web3.web3Instance
 
     // It's not requested? Let's get the drop on it so 
     // no other process requests it
     dispatch({type: SET_BLOCK_REQUESTED, number })
 
     // Now request the block and receipts for all its transactions
-    web3Request("getBlock", [number, true], provider, (err, block) => {
-      if (block.transactions.length == 0) {
-        return
-      }
+    let block = await web3Request("getBlock", [number, true], web3Instance)
```

Comment:
At this point, why aren't we just calling `await web3Instance.getBlock(number, true)`?

Reply:
Requests are cached when possible in the `requestCache.` In this case requests are definitely cached, since (99% of the time) blocks don't change. Though caching isn't all that important with a sandbox powering Ganache, it will become more important once forking comes along.

Reply:
Hmm, upon second thought looks like the cache will be hit via the provider. Maybe we don't need `web3Request` unless its a convenience (i.e., logging)

Reply:
@tcoulter I'd propose that perhaps the `RequestCache` is the wrong place to cache this data, then. Is it possible to have multiple reducers for a given action? If so, perhaps this should be happening when the `SET_BLOCK_NUMBER` action is dispatched, and the block should be chucked into the store?

Reply:
That's maybe not the best way to do that, because we don't want to eagerly cache _everything_ on a new block. However we could still use the store as a higher-level lazily-populated cache.

My other concern is that with Web 3 1.0 doing a bit more "magic" than I like, the pattern of relying on the provider to cache responses might be less efficient than it was with 0.22.x

Reply:
Anyway, back to my original point - calling `await web3Instance.getBlock(number, true)` goes through the RequestCache, as `web3Instance` is initialized with an instance of `ReduxWeb3Provider` as its provider.

Reply:
By `Hmm, upon second thought looks like the cache will be hit via the provider.` I was referring to our provider here:

https://github.com/trufflesuite/ganache/blob/369b3d5d756d1fabb809add28c1bbec8f3a0a50b/src/Actions/helpers/ReduxWeb3Provider.js#L16

You can see the caching mechanism here: https://github.com/trufflesuite/ganache/blob/develop/src/Reducers/RequestCache.js#L7

This is an important place to cache (as well as the easiest as far as I can tell) as it caches at the payload level. Some requests like `eth_getBlockByNumber` are hard cached -- i.e., not reset on a new block -- as there's no need to request the same block and its (potentially) few-hundred transactions more than once.

>  this is some idle musing

It's hard for me to tell what's important to respond to then! Hopefully we're not in violent agreement. 😃

Reply:
FYI: Sniper edited the comment above (in case you're reading by email)

Reply:
FWIW, violent agreement is my favorite kind of agreement. :laughing: